state/protocol/events.go

```diff
@@ -31,6 +31,8 @@ type Consumer interface {
 	// chain and its source of randomness is available).
 	// Formally, this callback is informationally idempotent. I.e. the consumer
 	// of this callback must handle repeated calls for the same block.
+	// TODO trigger this on block insertion (used to be on MarkValid)
+	//  - don't trigger for root block or below
 	BlockProcessable(block *flow.Header)
```

Comment:
## ⚠️ 
will this be fixed in a downstream PR? IIUC, other node roles depend on this callback for their business logic (like Verification nodes checking blocks). I think we are currently not calling this at all anymore.

Reply:
No, this should be fixed here!

I've moved this event to block insertion and updated the tests to check for this event more clearly: https://github.com/onflow/flow-go/pull/3303/commits/2a82b71aa8d86cdd32f8d8df90c59d6d659735b5. This was caught by unit tests, but broader-scope tests where it wasn't easy to see what was going wrong.

Reply:
Thanks for the fix. I noticed that  we are querying the root height for each inserted block twice, despite this being a constant. Please see [my comment below](https://github.com/onflow/flow-go/pull/3303#discussion_r1004980707)